sklearn/tests/test_common.py

```diff
@@ -204,3 +205,25 @@ def test_class_support_removed():
 
     with pytest.raises(TypeError, match=msg):
         parametrize_with_checks([LogisticRegression])
+
+
+def test_strict_mode_check_estimator():
+    # Make sure the strict checks are properly ignored when strict mode is off
+    # in check_estimator.
+    # We can't check the message because check_estimator doesn't give one.
+
+    with pytest.warns(SkipTestWarning):
+        # LogisticRegression has no _xfail_checks, but check_n_features_in is
+        # still skipped because it's a strict check
+        check_estimator(LogisticRegression(), strict_mode=False)
+
+    with pytest.warns(SkipTestWarning):
+        # NuSVC has some _xfail_checks. check_n_features_in is skipped along
+        # with the other checks in the tag.
+        check_estimator(NuSVC(), strict_mode=False)
+
+
+@parametrize_with_checks([LogisticRegression(), NuSVC()], strict_mode=False)
+def test_strict_mode_parametrize_with_checks(estimator, check):
+    # Not sure how to test parametrize_with_checks correctly??
```

Comment:
would it be possible to have a dummy estimator to check the `strict_mode={False, True}`? We probably shouldn't rely on the existing estimators here, we may end up fixing them :P

Reply:
These dummy estimators that we'd create would still need to depend on some of our builtin estimators, otherwise the checks won't pass. So wouldn't that just shift the issue?